### PR TITLE
[WIP] Adds auth with agents and accounts.

### PR DESCRIPTION
**Pull request by pietgeursen**

### CODE REVIEW

File: service.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  const app = feathers()
+  app.configure(configuration())
+  app.use(bodyParser.json())
+  app.use(bodyParser.urlencoded({ extended: true }))
+  app.configure(rest())
+  app.configure(hooks())
+  // services
+  forEach(services, (service, name) => {
+    app.use(name, service(db))
+    app.service(name).after(
```
Comment (ahdinosaur) [inline service.js]: this is cool!

File: client.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import auth from 'feathers-authentication-client'
+
+const localStorage = window ? window.localStorage : null
+
+const client = feathers()
+  .configure(rest('/api').fetch(window.fetch.bind(window)))
+  .configure(hooks())
+  .configure(auth({
+    storage: localStorage,
+    accessTokenKey: '41S'
```
Comment (ahdinosaur) [inline client.js:13]: dog dog dog dog dog dog dog dog

Comment (ahdinosaur) [reply client.js:13]: or does this even need to be configured?

File: package.json [hunk 3]
```diff
@@ -1,14 +1,84 @@
 {
-  "name": "dogstack",
-  "version": "0.0.5",
-  "description": "a popular-choice grab-bag framework for teams working on production web apps",
+  "name": "dogstack-example-dogs",
```
Comment (ahdinosaur) [inline package.json:2]: dogstack-example?

File: README.md [hunk 4]
```diff
@@ -16,6 +16,14 @@
   :cat: see also <a href='https://github.com/enspiral-root-systems/cat-stack'>catstack</a>, dogstack's smarter, slimmer, more cunning partner in crime
 </h4>
 
+-> make topic called agent
```
Comment (ahdinosaur) [inline README.md:19]: nice, thanks!

**Review by ahdinosaur — APPROVED** [review]

**Comment by ahdinosaur** [comment]
merge merge merge merge

File: client.js (hunk 2 above)
Comment (pietgeursen) [reply client.js:13]: dogsdogsdogs. done

File: package.json (hunk 3 above)
Comment (pietgeursen) [reply package.json:2]: yep. done

File: service.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      before: service.before || {},
+      after: service.after || {},
+      error: service.error || {}
+    })
+  })
+
+  app.configure(authentication(app.get('auth')))
+    .configure(jwt())
+    .configure(local())
+    .configure(authService)
```
Comment (ahdinosaur) [inline service.js:40]: @pietgeursen just wondering, what does this do?

Comment (ahdinosaur) [reply service.js:40]: nevermind, i see it's kinda how it all works. :sweat_smile:

**Pull request merged**